Cannot find module '@next/third-parties/google' when trying to import GoogleTagManager

### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/musing-solomon-f4tnqz?file=%2Fapp%2Flayout.tsx%3A8%2C11

### To Reproduce

1. Start application with next dev
2. Check layout.tsx
3. import error `Cannot find module '@next/third-parties/google' or its corresponding type declarations.typescript(2307)`

### Current vs. Expected behavior

Expected: import work as intended in documentation https://nextjs.org/docs/app/building-your-application/optimizing/third-party-libraries#google-tag-manager

Current: import not working
![image](https://github.com/vercel/next.js/assets/62075475/6baaf911-5196-4cdd-984c-62c86c306c31)



### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Node: v20.9.0
Next: 14.0.3,
React: 8.2.0
react-dom: 18.2.0,
npm: 10.1.0

Windows 11
```


### Which area(s) are affected? (Select all that apply)

App Router

### Additional context

Tested using next/canary, next v14 and nextv13, all with same problems.

after changing the import to  `import { GoogleTagManager } from '@next/third-parties/dist/google'`, the error goes away but when building the application i got an error because there is no exported module in node_modules\@next\third-parties\dist\google.

package.json

![image](https://github.com/vercel/next.js/assets/62075475/f0670b24-9091-4857-bd53-74d32cba0727)


Error message:

Failed to compile.
./src/app/layout.tsx
Module not found: Package path ./dist/google is not exported from package /tmp/build_6a78ac44/node_modules/@next/third-parties (see exports field in /tmp/build_6a78ac44/node_modules/@next/third-parties/package.json)

--- a/packages/third-parties/google.d.ts
+++ b/packages/third-parties/google.d.ts
@@ -1,0 +1,1 @@
+export * from './dist/types/index'

--- a/packages/third-parties/package.json
+++ b/packages/third-parties/package.json
@@ -6,9 +6,11 @@
     "directory": "packages/third-parties"
   },
   "exports": {
-    "./google": "./dist/google/index.js"
+    "./google": {
+      "types": "./dist/google/index.d.ts",
+      "default": "./dist/google/index.js"
+    }
   },
-  "types": "dist/types/google.d.ts",
   "files": [
     "dist"
   ],

